### ISSUE TITLE
Common X12 Segments

### ISSUE THREAD

**Issue by michaelachrisco** [kenwilcox/X12Parser#1]
First of all, great job on the library! 

I am just playing around with the library and comparing it to other common X12 parsing/generating libraries such as EDI.NET, edifabric, etc.. 

Ill start with on of the most common segments, the DTP segment. Should have a MR soon. How should I go about testing for segments?

**Comment by kenwilcox** [comment]
Thanks for your desire to contribute.

I'm not sure what you mean by testing for segments. I currently only need to parse 835 files. I have around 350 files and I believe I have all those segments. I did find a 277 file that I was going to go through next and add those missing ones. I was using the tester to figure out which segments default to just the X12 base class and going from there.

I added a [LINQPad](http://www.linqpad.net/) script that can take the Ruby hippo project segments and converts them to a "lame" CSharp segment. Not all of them were a correct conversion, there's some bits in the Ruby stuff that I'm not sure what it's doing, so there's still some manual adjustments necessary. 

If you don't have Ruby or Gems installed I can add the 100 or so segments from the hippo project to this one for conversion assistance if you would be interested in helping with that.

What X12 documents are you trying to parse?

Thanks again, really.

**Comment by michaelachrisco** [comment]
TIL about LINQPag. Interesting, never heard of it! Before I did C#, I had a background in Ruby so feel free to put up the translated Ruby code should you feel like diving into it! I am not familiar with hippo but I did use the  StupidEdI gem to validate a couple of files. Its slow but does the job. 

ATM I am just playing around with the library. I'm looking to unit test the files in some way/shape/form. I believe EDI.Net has a good method of testing the files as a whole with all Segments/Loops in one file that gets compared with what you expect. I see you are doing something similar with the ISA segment?

The most common X12 EDI file I get is the 834 which is worked a variety of ways. I will either validate, then parse the elements(with accompanying 999) into a staging server or generate the file and sent if off to its respective client/vendor. Most of my recent work is just generating and sending the files themselves. I've worked with most of the healthcare files (like the 834/835/275/276/277/etc...) always just looking for new solutions in the EDI world!
